antsibull/dependency_files.py

```diff
@@ -155,7 +155,19 @@ def write(self, ansible_version: 'PypiVer', ansible_base_version: str,
         """
         records = []
         for dep, version in dependencies.items():
-            records.append(f'{dep}: >={version.major}.{version.minor}.0,<{version.next_major()}')
+            if version.prerelease and version.minor == 0 and version.patch == 0:
+                # Prereleases of major releases (ex: 2.0.0-b1) are a special case.  They sort
+                # before their major number so next_major() does something different (it would
```

Comment:
Isn't this also the case for pre-releases of minor versions? If `2.1.0-b1` is the latest release, the code will currently construct the range `>=2.1,<3.0.0`, which is empty, since `2.1.0-b1 < 2.1.0`.

Reply:
Yeah, you're right.